js/src/redux/providers/chainMiddleware.js

```diff
@@ -24,20 +22,34 @@ export default class ChainMiddleware {
       if (action.type === 'statusCollection') {
         const { collection } = action;
 
-        if (collection && collection.netChain) {
-          const newChain = collection.netChain;
+        if (collection) {
           const { nodeStatus } = store.getState();
+          const { netChain, nodeKind } = nodeStatus;
+          const newChain = collection.netChain;
+          const newNodeKind = collection.nodeKind;
+          let reloadChain = false;
+          let reloadType = false;
+
+          // force reload when chain has changed and is not initial value
+          if (newChain) {
+            const hasChainChanged = newChain !== netChain;
+            const isInitialChain = netChain === DEFAULT_NETCHAIN;
+
+            reloadChain = !isInitialChain && hasChainChanged;
```

Comment:
I don't really follow why a reload does not happen if current chain is a default chain? (I know the logic here didn't change, just curious).